packages/components/pie-modal/src/index.ts

```diff
@@ -44,20 +44,59 @@ export class PieModal extends RtlMixin(LitElement) {
                         variant="ghost-tertiary"
                         class="c-modal-closeBtn"></pie-icon-button>
                 </header>
-                <article>
-                    <div class="c-modal-content">
-                        <slot></slot>
-                    </div>
+                <article class="c-modal-content">
+                    <slot></slot>
                 </article>
             </dialog>
         `;
     }
 
+    updated () {
+        this._handleDialogLightDismiss();
+    }
+
+    /**
+     * We require this because toggling the prop `isOpen` itself won't
+     * allow the dialog to open in the correct way (with the default background),
+     * the method `showModal()` needs to be invoked.
+     *
+     * https://lit.dev/docs/components/lifecycle/#willupdate
+     *
+     * @param changedProperties
+     */
+    willUpdate (changedProperties: Map<string, any>) {
```

Comment:
as a rule of thumb we should avoid using `any` in our code. [My pr](https://github.com/justeattakeaway/pie/blob/32c9184d60a517e7756ad64697dd9780f9a56c08/packages/components/pie-modal/src/index.ts#L34) proposes a way to avoid using `any` in this particular situation. You're welcome to add that to this PR instead if you want to get this in first?

Reply:
Yes I did think about that but wasn't sure as i took this directly from LIT. Will have a look 👍

Reply:
I've updated it so there would be a type checking now.